### ISSUE TITLE
dependabot should not be active in forks

### ISSUE THREAD

**Issue by return42** [searxng/searxng#155]
**Is your feature request related to a problem? Please describe.**

The dependabot creates PRs and noise in forks where it is not needed or wanted. E.g. in my [fork](https://github.com/return42/searxng/search?q=dependabot) the issues from dependabot are unwanted and it is not only mine where this is useless

**Describe the solution you'd like**

Enable [github/dependabot.yml](https://github.com/searxng/searxng/blob/master/.github/dependabot.yml) only for the _origin_ https://github.com/searxng/searxng/

**Describe alternatives you've considered**

It seems we are not alone with the problem and a solution is not shine at  nearer horizon ...

- https://github.com/dependabot/dependabot-core/issues/2804#issuecomment-769954600
- https://github.com/dependabot/dependabot-core/issues/2198

~~For the interim we could deactivate dependabot and update manually.~~

I think we have ~~to act here~~ to wait until https://github.com/dependabot/dependabot-core/issues/2804 is fixed, until this is fixed ~~because~~ each fork has a master branch that we cannot change after the fork is created / these dependabots from forks will run infinite.

**Comment by dalf** [comment]
Perhaps my memory tricks me.

Dependabot will be triggered:
* for forks that have enabled dependabot.
* or unconditionally, on repositories that have been forked before dependabot was available.

The second case is problematic, but it won't happen here.
For example https://github.com/unixfox/searxng/pulls doesn't have any PR from dependabot.

Moreover, I wish to enable dependabot (or whatever that creates PR) on javascript dependencies:
https://github.com/dalf/searxng/blob/master/.github/dependabot.yml

**Comment by return42** [comment]
Is this issue really solved? .. #235 does not solve this issue.

**Comment by dalf** [comment]
#235 doesn't solve the issue.

The issue is: on a fork, if dependabot is enabled, there is no way to disable it.

Possible alternative:
* manual check the dependencies: do you have time to check the Python and NPM dependencies every week, every week of the year? Who does the check? Do we take turn? If there is a CVE on one of the dependency, dependabot will push the information (a new PR). A manual check, not pulled (open a terminal, upgrade, git diff, etc...).
* manual check from time to time with possible out dated and/or CVE of dependencies: I'm "meh" about this.
* implement a custom solution:
  * at least to display the out of date dependencies (where? how?)
  * to create different PR (light dependabot).

> For the interim we could deactivate dependabot and update manually.

GitHub has known about this for 2 years

IMO, dependabot is good compromise for the current size of the project.

---

May be there is a way to crash dependabot script with a config that works only on this repository, for example: add a label that exists only this repository.

**Comment by return42** [comment]
>  The issue is: on a fork, if dependabot is enabled, there is no way to disable it.

Yes we have to wait until https://github.com/dependabot/dependabot-core/issues/2804 is fixed.

> IMO, dependabot is good compromise for the current size of the project.

Sorry, I was very misleading at this point ... forget this sentence:

> ~~For the interim we could deactivate dependabot and update manually.~~

It was not my aim to deactivate dependabot (even if this is the only alternative right now) / I think dependabot does a good job for us at SearXNG.

I just wanted to left this issue open until https://github.com/dependabot/dependabot-core/issues/2804 has been fixed.

I updated the initial https://github.com/searxng/searxng/issues/155#issue-924700185 / again sorry my initial comment was very misleading.
